src/output.rs

```diff
@@ -2,67 +2,169 @@ use self::errors::*;
+            StrPredicate::Contains(ref pred) => pred.verify_str(got),
+        }
+    }
+}
 
-    pub fn execute(&self, output: &Output, cmd: &[OsString]) -> super::errors::Result<()> {
-        let observed = String::from_utf8_lossy(self.kind.select(output));
+/// Assertions for command output.
+#[derive(Debug, Clone)]
+pub struct Output {
+    pred: StrPredicate,
+}
 
-        let result = if self.fuzzy {
-            self.matches_fuzzy(&observed)
-        } else {
-            self.matches_exact(&observed)
+impl Output {
+    /// Expect the command's output to **contain** `output`.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// extern crate assert_cli;
+    ///
+    /// assert_cli::Assert::command(&["echo"])
+    ///     .with_args(&["42"])
+    ///     .stdout(assert_cli::Output::contains("42"))
+    ///     .unwrap();
+    /// ```
+    pub fn contains<O: Into<String>>(output: O) -> Self {
+        let pred = ContainsPredicate {
+            expect: output.into(),
+            expected_result: true,
         };
-        result.map_err(|e| {
-            super::errors::ErrorKind::OutputMismatch(cmd.to_vec(), e, self.kind)
-        })?;
+        Self::new(StrPredicate::Contains(pred))
+    }
 
-        Ok(())
+    /// Expect the command to output **exactly** this `output`.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// extern crate assert_cli;
+    ///
+    /// assert_cli::Assert::command(&["echo"])
+    ///     .with_args(&["42"])
+    ///     .stdout(assert_cli::Output::is("42"))
+    ///     .unwrap();
+    /// ```
+    pub fn is<O: Into<String>>(output: O) -> Self {
+        let pred = IsPredicate {
+            expect: output.into(),
+            expected_result: true,
+        };
+        Self::new(StrPredicate::Is(pred))
+    }
+
+    /// Expect the command's output to not **contain** `output`.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// extern crate assert_cli;
+    ///
+    /// assert_cli::Assert::command(&["echo"])
+    ///     .with_args(&["42"])
+    ///     .stdout(assert_cli::Output::doesnt_contain("73"))
+    ///     .unwrap();
+    /// ```
+    pub fn doesnt_contain<O: Into<String>>(output: O) -> Self {
+        let pred = ContainsPredicate {
+            expect: output.into(),
+            expected_result: false,
+        };
+        Self::new(StrPredicate::Contains(pred))
+    }
+
+    /// Expect the command to output to not be **exactly** this `output`.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// extern crate assert_cli;
+    ///
+    /// assert_cli::Assert::command(&["echo"])
+    ///     .with_args(&["42"])
+    ///     .stdout(assert_cli::Output::isnt("73"))
+    ///     .unwrap();
+    /// ```
+    pub fn isnt<O: Into<String>>(output: O) -> Self {
+        let pred = IsPredicate {
+            expect: output.into(),
+            expected_result: false,
+        };
+        Self::new(StrPredicate::Is(pred))
+    }
```

Comment:
After reading for of these I fell the need to write a macro for this… which would probably not make the code much clearer 😅